HW05_part_B.Rmd

```diff
@@ -0,0 +1,173 @@
+  # adjust labels
+  labs(title = "Sensitivity of different cancer types to the compound vorinostat", x = "Cancer Type", y = "Mean Area Under Curve")
+  
+  
+
+
+```
+
+**Additionally, we can calculate the mean AUC values for each cancer type and use the resulting values as basis for comparison:**
+
+``` {r}
+
+# create new data frame with mean AUC for each cancer_type
+auc_low_vor2 <- auc_low_vor %>%
+  group_by(cancer_type) %>%
+  summarize(mean_auc = mean(area_under_curve)) %>% 
+  arrange(mean_auc)
+
+# generate bar graph comparing the mean AUC between cancer types
+ggplot(auc_low_vor2, aes(x = reorder(cancer_type, mean_auc), y = mean_auc)) +
+  geom_col(fill = "dodgerblue4", alpha = 0.5, color = "black") +
+  # add label inside the bars, it is very messy to include them below the x-axis
+  geom_text(aes(label = cancer_type), position = position_stack(vjust = 0.5), angle = 90, size = 3) +
+  # adjust theme
+  theme_classic() +
+  theme(axis.text.x = element_blank(), axis.ticks.x = element_blank()) +
+  # adjust labels
+  labs(title = "Sensitivity of different cancer types to the compound vorinostat", subtitle = "Based on mean AUC values", x = "Cancer Type", y = "Mean Area Under Curve")
+  
+# print the name of the cancer type with the lowest mean AUC to vorinostat
+print(paste("The cancer type with the lowest mean AUC values to the compound vorinostat is", auc_low_vor2[1,1], "."))
+
+```
+
+
+### Which compound is the prostate cancer cell line 22RV1 most sensitive to? (For 22RV1, which compound has the lowest AUC value?)
+
+```{r}
+
+# generate new data frame
+auc_low_prostate <- data_complete %>%
+  select(area_under_curve, ccl_name, cpd_name) %>%
+  filter(ccl_name == "22RV1") %>%
+  arrange(area_under_curve) 
+  
+
+
+head(auc_low_prostate)
+
+
+```
+
+**Note: When trying to filter for the 22RV1 cell line in my data frame, I receive an error message, saying that there is no data in my data frame for that cell line. I am not sure if anything's wrong with my data frame, but I am using the VCAP prostate cancer cell line instead for this question.**
+
+
+```{r}
+
+# generate alternative data frame, filtering for VCAP cell line instead of 22RV1
+auc_low_prostate2 <- data_complete %>%
+  select(area_under_curve, ccl_name, cpd_name) %>%
+  filter(ccl_name == "VCAP") %>%
+  arrange(area_under_curve) 
+  
+
+
+print(paste("The prostate cancer cell line VCAP is most sensitive to the compound", auc_low_prostate2[1, 3], "."))
+
+
+```
+
+
+### For the 10 compounds that target EGFR, which of them has (on average) the lowest AUC values in the breast cancer cell lines?
+
+```{r}
+
+# generate new data frame 
+auc_egfr <- data_complete %>%
+  select(area_under_curve, cancer_type, gene_symbol_of_protein_target, cpd_name) %>%
+  filter(str_detect(gene_symbol_of_protein_target, "EGFR"), cancer_type == "breast") %>%
+  group_by(cpd_name) %>%
+  summarize(mean_auc = mean(area_under_curve)) %>%
+  arrange(mean_auc)
```

Comment:
A small question only for discussion. Should we calculate the mean AUC in each cell line first and than calculate the mean of all breast cancer cell lines for each compound, to make sure that all cell lines weight the same? This will not influence much at least in this case, and I just want to discuss it.

Reply:
That's an interesting question! Are you thinking to take the mean AUC for each compound/cell line pair first, and then use that initial mean to calculate the mean AUC for all breast cancer cell lines per compound?
Personally, I think that would be a good idea, especially when working with a dataset that has some variability in the number of observations per cell line/compound pair. If that's the case, just taking the mean for all breast cancer cell lines could "push" it towards the results for the cell lines with more observations. However, I would expect it to be less of an issue when working with a dataset that has the same number of observations for each cell line/compound pair. 
It's an interesting question, and I am happy to discuss it more. What do you think?

Reply:
Personally I calculated the mean AUC for each compound/cell line pair first, but later when seeing your code I tried to compare the number of observations and the number of compound/cell line pairs (398 and 353, respectively), I noticed that most pairs have only one observation, thus this step will not influence too much in this case. Though I still think it is more precise to calculate the average in two steps, and it is not too difficult in this case.
But if dealing with a large datasets with more different conditions, for example, the cancer types are different, the compound concentrations used to treat the cells are different, or the treating times are different (just for example and I know it not proper to compute the average for these variables), is it necessary to calculate the average step by step, or we can just mix the observations together and calculate at only one time? It might greatly save workloads, if it will not be too less precise.